### DISCUSSION TITLE
How to remove share/nvim/runtime/queries dir from runtimepath?

### DISCUSSION THREAD

**Discussion by Shaobin-Jiang** [nvim-treesitter/nvim-treesitter#6598] [Q&A]
I am using Windows 11 and when I do `checkhealth nvim-treesitter`, I get this following error:

```
Parser/Features         H L F I J
  - c                   ✓ ✓ ✓ ✓ ✓
  - c_sharp             ✓ ✓ ✓ . ✓
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - css                 ✓ . ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - json                ✓ ✓ ✓ ✓ .
  - lua                 ✓ ✓ ✓ ✓ x
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

The following errors have been detected: ~
- ERROR lua(injections): .../current/share/nvim/runtime/lua/vim/treesitter/query.lua:259: query: invalid structure at position 3272 for language lua
  lua(injections) is concatenated from the following files:
  | [ERROR]:"C:/Users/dell/AppData/Local/nvim-data/lazy/nvim-treesitter/queries/lua/injections.scm", failed to load: .../current/share/nvim/runtime/lua/vim/treesitter/query.lua:259: query: invalid structure at position 3272 for language lua
```

I am aware that a solution to this sort of issue can be viewed [here](https://github.com/nvim-treesitter/nvim-treesitter/issues/3092). However, I for one find that lacking in details.

Now, with `share/nvim/runtime/` in `runtimepath`, I cannot prevent this issue from occurring. My solution for now is to remove the `share/nvim/runtime/queries` manually, but that certainly is not a wise option. Is there a way to remove it only from `runtimepath` without actually deleting the directory? Or is there some other alternative solutions to the error in health check?

**Comment by clason** [comment]
No, there is not. Nvim-treesitter 1.0 (`main` branch) will improve on that, but for now if you install nvim-treesitter, you _must_ install the bundled parsers to make sure queries match.

**Reply by Shaobin-Jiang** [reply]
Do you mean these five parsers: c / lua / vim / vimdoc / query? I have installed them and the error persists.

**Reply by clason** [reply]
That depends on your Neovim version, yes. If the error persist, you may have outdated parsers lying around somewhere, and the pinned issue gives you the tools to track them down. As this is a config issue, we can't do more to help you.
